docs/pages/docs/guides/cohere.mdx

```diff
@@ -0,0 +1,119 @@
+---
+title: Cohere
+---
+
+# Cohere
+
+import { Steps, Callout } from 'nextra-theme-docs'
+
+Vercel AI SDK provides a set of utilities to make it easy to use Cohere's API. In this guide, we'll walk through how to use the utilities to create a text completion app.
+
+## Guide: Text Completion
+
+<Steps>
+
+### Create a Next.js app
+
+Create a Next.js application and install `ai`:
+
+```sh
+pnpm dlx create-next-app my-ai-app
+cd my-ai-app
+pnpm install ai
+```
+
+### Add your Cohere API Key to `.env`
+
+```env filename=".env"
+COHERE_API_KEY=xxxxxxx
+```
+
+### Create a Route Handler
+
+Create a Next.js Route Handler that uses the Edge Runtime to generate a response to a series of messages via Cohere's API, and returns the response as a streaming text response.
+
+For this example, we'll create a route handler at `app/api/completion/route.ts` that accepts a `POST` request with a `prompt` string:
+
+```tsx filename="app/api/completion/route.ts" showLineNumbers
+import { StreamingTextResponse, CohereStream } from 'ai'
+
+// IMPORTANT! Set the runtime to edge
+export const runtime = 'edge'
+
+export async function POST(req: Request) {
+  // Extract the `prompt` from the body of the request
+  const { prompt } = await req.json()
+
+  const body = JSON.stringify({
+    prompt,
+    model: 'command-nightly',
+    stream: true
+  })
+
+  const response = await fetch('https://api.cohere.ai/v1/generate', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${process.env.COHERE_API_KEY}`
+    },
+    body
+  })
+
+  const reader = response.body?.getReader()!
+
+  // Extract the text response from the Cohere stream
+  const stream = CohereStream(reader)
```

Comment:
Can we keep this consistency with the other Stream wrappers and pass in the entire response?

Reply:
sure, done! :)

Reply:
hey @MaxLeiter, 

I also added a new helper function, `makeTextLineIterator` which prevents some pitfalls for long responses. Indeed is not necessary for short responses, but I got half of the chunk when I tested with long responses when I was iterating over `await reader.read()`. 

If you folks have found something else to prevent this issue, let me know :)